docs/flow/modin/core/execution/ray/implementations/pandas_on_ray/index.rst

```diff
@@ -1,8 +1,34 @@
+PandasOnRay Execution
+=====================
+
+Queries that perform data transformation, data ingress or data egress using the `pandas on Ray` execution
+pass through the Modin components detailed below.
+
+To enable the `pandas on Ray` execution, please refer to the usage section in :doc:`pandas on Ray </UsingPandasonRay/index>`.
+
+Data Transformation
+'''''''''''''''''''
+
+.. image:: /img/pandas_on_ray_data_transform.svg
+   :align: center
+
+When a user calls any :py:class:`~modin.pandas.dataframe.DataFrame` API, a query starts forming at the `API` layer
+to be executed at the `Execution` layer. The `API` layer is responsible for processing the query appropriately,
+for example, determining whether the final result should be a ``DataFrame`` or ``Series`` object, and sanitizing the input to the
+:py:class:`~modin.core.storage_formats.pandas.query_compiler.PandasQueryCompiler`, e.g. validating a parameter from the query
+and defining specific intermediate values to provide more context to the query compiler.
+The :py:class:`~modin.core.storage_formats.pandas.query_compiler.PandasQueryCompiler` is responsible for
+processing the query, received from the :py:class:`~modin.pandas.dataframe.DataFrame` `API` layer,
+to determine how to apply it to a subset of the data - either cell-wise or along an axis-wise partition backed by the `pandas`
+storage format and mapping the query to one of the :doc:`Core Operators </flow/modin/core/dataframe/algebra>` of
+:py:class:`~modin.core.execution.ray.implementations.pandas_on_ray.dataframe.dataframe.PandasOnRayDataframe` that inherits
+generic functionality from ``GenericRayDataframe`` and :py:class:`~modin.core.dataframe.pandas.dataframe.dataframe.PandasDataframe`.
```

Comment:
We do not have `GenericRayDataframe` class.

Reply:
But we have doc, which I mentioned above with description of `GenericRayDataframePartitionManager`. It can help with fast finding info about `GenericRayDataframe` is mean in that context (becaise it isn't a class).

Reply:
`GenericRayDataframe` is a class here and I would prefer to have a direct link to it once we add an implementation of the class.

Reply:
Ok. In this case we should don't forget to do this (mention of an issue related to adding of `GenericRayDataframe` in the comment in this thread will be enough)

Reply:
Added todo.